emu_sv/sv_config.py

```diff
@@ -115,21 +117,40 @@ def _expected_kwargs(self) -> set[str]:
 
     def monkeypatch_observables(self) -> None:
         obs_list = []
+
         for _, obs in enumerate(self.observables):  # monkey patch
             obs_copy = copy.deepcopy(obs)
+
             if isinstance(obs, Occupation):
-                obs_copy.apply = MethodType(  # type: ignore[method-assign]
-                    qubit_occupation_sv_impl, obs_copy
-                )
-            elif isinstance(obs, EnergyVariance):
-                obs_copy.apply = MethodType(  # type: ignore[method-assign]
-                    energy_variance_sv_impl, obs_copy
-                )
+                if self.noise_model.noise_types == ():
+                    obs_copy.apply = MethodType(  # type: ignore[method-assign]
+                        qubit_occupation_sv_impl, obs_copy
+                    )
+                else:
+                    obs_copy.apply = MethodType(  # type: ignore[method-assign]
+                        qubit_occupation_sv_den_mat_impl, obs_copy
+                    )
+            if isinstance(obs, EnergyVariance):
+                if self.noise_model.noise_types == ():
+                    obs_copy.apply = MethodType(  # type: ignore[method-assign]
+                        energy_variance_sv_impl, obs_copy
+                    )
+                else:
+                    raise Exception("Not implemented for density matrix")
```

Comment:
you can make these blocks shorter with smth like:
```python
if self.noise_model.noise_types != ():
  raise Exception("Not implemented for density matrix")
obs_copy.apply = MethodType(  # type: ignore[method-assign]
                        energy_variance_sv_impl, obs_copy
                    )
```

because `raise` interrupts the further code execution